### DISCUSSION TITLE
Running out RAM - Noob question

### DISCUSSION THREAD

**Discussion by klausnielsen** [tttapa/Control-Surface#1006] [General]
Hello everyone - Hello Pieter
I'm learning to use your super cool library. Working on a Nano, I'm running out of RAM in a sketch that uses 8 multiplexers and array for 64 CCPotentiometers. (Will probably have to upgrade to Mega or similar, since the Nano is low on memory.)
I'm trying to use the [tips for saving memory](https://tttapa.github.io/Control-Surface-doc/Doxygen/da/dc1/FAQ.html#faq-save-memory), but it doesn't have an effect. Am I missing some additional lines in order for them to work?
```
#include <Control_Surface.h>

#define IGNORE_SYSEX 0
#define NO_SYSEX_OUTPUT 0
#define CS_TRUE_CONTROL_SURFACE_INSTANCE 1
#define DISABLE_PIPES 0

auto &serial = Serial;
```
Any comments or help would be super appreciated
Cheers
Klaus

**Comment by tttapa** [comment]
You have to change the values in the Settings.hpp file in your Control Surface installation (inside of the ~/Arduino/libraries folder). Defining those macros inside of the sketch cannot have any effect.

**Comment by klausnielsen** [comment]
hey - it worked! :) thanks a bunch
and thanks for the quick response

now I might just be able to squeeze the sketch on to a Nano :)
